Documentation/ReadStores.md

```diff
@@ -0,0 +1,109 @@
+# Read model stores
+
+In order to create query handlers that perform and enable them search across
+multiple fields, read models or projects are used.
+
+Read models are a flatten views of a subset or all aggregate domain events
+created specifically for efficient queries.
+
+Here's a simple example of how a read models for doing searches for usernames
+could look. The read model handles the `UserCreated` domain event event to get
+the username and user ID.
+
+```csharp
+public class UserReadModel : IReadModel,
+  IAmReadModelFor<UserAggregate, UserId, UserCreated>
+{
+  public string UserId { get; set; }
+  public string Username { get; set; }
+
+  public void Apply(
+    IReadModelContext context,
+    IDomainEvent<UserAggregate, UserId, UserCreated> domainEvent)
+  {
+    UserId = domainEvent.AggregateIdentity.Value;
+    Username = domainEvent.AggregateEvent.Username.Value;
+  }
+}
+```
+
+## Read model locators
+
+Typically the ID of read models are the aggregate identity, but sometimes this
+isn't the case. Here are some examples.
+
+- Items from a collection on the aggregate root
+- Deterministic ID created from event data
+- Entity within the aggregate root
```

Comment:
- root

Comment: referring to `aggregate root` which is conceptually an entity could be confusing or is the intention to state that an entity can have an entity as dependency?